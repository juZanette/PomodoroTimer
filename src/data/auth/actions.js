import { firebase } from '../../firebase/firebase'

export const setUserInfo = (userInfo) => ({
  type: 'SET_USER_INFO',
  userInfo,
})

export const linkAccount = (authProvider) => {
  return async (dispatch) => {
    try {
      const res = await firebase.auth().currentUser.linkWithPopup(authProvider)

      const user = res.user

      const userInfo = { uid: user.uid }

      userInfo.creationTime = user.metadata.creationTime

      if (user.providerData && user.providerData.length) {
        userInfo.name = user.providerData[0].displayName
        userInfo.photo = user.providerData[0].photoURL
      }

      dispatch(setUserInfo(userInfo))
    } catch (e) {
      if (e.code === 'auth/credential-already-in-use') {
        await firebase.auth().signInWithCredential(e.credential)
      }
    }
  }
}

export const setCreationTime = (creationTime) => ({
  type: 'SET_CREATION_TIME',
  creationTime,
})
