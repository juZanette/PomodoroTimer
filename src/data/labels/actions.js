import fs from '../../firebase/firebase'

export const setFullscreenDialog = (fullscreenDialog) => ({
  type: 'SET_FULLSCREEN_DIALOG',
  fullscreenDialog,
})

export const setDesktopDialog = (desktopDialog) => ({
  type: 'SET_DESKTOP_DIALOG',
  desktopDialog,
})

export const setDeleteAlert = (deleteAlert) => ({
  type: 'SET_DELETE_ALERT',
  deleteAlert,
})

export const setMenuOpened = (menuOpened) => ({
  type: 'SET_MENU_OPENED',
  menuOpened,
})

export const setFormValue = (formValue) => ({
  type: 'SET_FORM_VALUE',
  formValue,
})

export const setLabelEditting = (labelEditting) => ({
  type: 'SET_LABEL_EDITTING',
  labelEditting,
})

export const setLabelSelected = (labelSelected) => ({
  type: 'SET_LABEL_SELECTED',
  labelSelected,
})

export const addLabel = (label) => ({
  type: 'ADD_LABEL',
  label,
})

export const startAddLabel = (label) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    const ref = await fs.collection(`users/${uid}/labels`).add(label)

    dispatch(
      addLabel({
        id: ref.id,
        ...label,
      })
    )

    return ref
  }
}

export const editLabel = (id, updates) => ({
  type: 'EDIT_LABEL',
  id,
  updates,
})

export const startEditLabel = (id, updates) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    await fs.doc(`users/${uid}/labels/${id}`).update({
      ...updates,
    })

    dispatch(editLabel(id, updates))
  }
}

export const deleteLabel = (id) => ({
  type: 'DELETE_LABEL',
  id,
})

export const startDeleteLabel = (id) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid
    await fs.doc(`users/${uid}/labels/${id}`).delete()

    dispatch(deleteLabel(id))
  }
}

export const setLabels = (labels) => ({
  type: 'SET_LABELS',
  labels,
})

export const startSetLabels = () => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    const labelsRef = await fs.collection(`users/${uid}/labels`).get()

    const labels = labelsRef.docs.map((label) => ({
      id: label.id,
      ...label.data(),
    }))

    dispatch(setLabels(labels))

    return labelsRef
  }
}
