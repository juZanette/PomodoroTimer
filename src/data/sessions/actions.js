import fs from '../../firebase/firebase'

export const setSessions = (sessions) => ({
  type: 'SET_SESSIONS',
  sessions,
})

export const startSetSessions = () => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    const sessionsRef = await fs.collection(`users/${uid}/sessions`).get()

    const sessions = sessionsRef.docs.map((session) => ({
      id: session.id,
      ...session.data(),
    }))

    dispatch(setSessions(sessions))

    return sessionsRef
  }
}

export const addSession = (session) => ({
  type: 'ADD_SESSION',
  session,
})

export const startAddSession = (session) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid
    const { duration, label = null, createdAt } = session

    const ref = await fs.collection(`users/${uid}/sessions`).add({
      duration,
      label,
      createdAt,
    })

    dispatch(
      addSession({
        id: ref.id,
        ...session,
      })
    )

    return ref
  }
}
