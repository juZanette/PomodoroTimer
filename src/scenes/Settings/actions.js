export const setRounds = roundsNumber => ({
  type: 'SET_ROUNDS',
  roundsNumber,
})
