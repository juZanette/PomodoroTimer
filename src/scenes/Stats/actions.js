export const setRounds = () => ({
  type: 'INCREMENT_TOTAL_SESSIONS',
})
