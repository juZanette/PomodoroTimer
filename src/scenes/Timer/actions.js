export const setStatus = status => ({
  type: 'SET_STATUS',
  status,
})

export const setProgress = progress => ({
  type: 'SET_PROGRESS',
  progress,
})

export const setTimeLeft = timeLeft => ({
  type: 'SET_TIME_LEFT',
  timeLeft,
})

export const saveInterval = interval => ({
  type: 'SAVE_INTERVAL',
  interval,
})

export const pauseTimer = () => ({
  type: 'PAUSE_TIMER',
})

export const resetTimer = () => ({
  type: 'RESET_TIMER',
})

export const setNextTimer = () => ({
  type: 'SET_NEXT_TIMER',
})
