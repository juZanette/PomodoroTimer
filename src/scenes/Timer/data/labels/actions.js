export const setDialogOpened = (dialogOpened) => ({
  type: 'SET_DIALOG_OPENED',
  dialogOpened,
})

export const setAlertOpened = (alertOpened) => ({
  type: 'SET_ALERT_OPENED',
  alertOpened,
})

export const setLabelToEdit = (labelToEdit) => ({
  type: 'SET_LABEL_TO_EDIT',
  labelToEdit,
})

export const setFormValue = (formValue) => ({
  type: 'SET_FORM_VALUE',
  formValue,
})
