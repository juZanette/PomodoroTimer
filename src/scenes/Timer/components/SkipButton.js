import React from 'react'
import IconButton from '@material-ui/core/IconButton'
import SkipNext from '@material-ui/icons/SkipNext'
import { useDispatch, useSelector } from 'react-redux'
import { setNextTimer } from '../actions'

const SkipButton = () => {
  const dispatch = useDispatch()
  const { timeLeft } = useSelector((state) => state.timer)
  const settings = useSelector((state) => state.settings)

  return (
    <IconButton
      disabled={!timeLeft}
      aria-label="Skip current timer"
      onClick={() => dispatch(setNextTimer(settings))}
    >
      <SkipNext />
    </IconButton>
  )
}

export { SkipButton }
