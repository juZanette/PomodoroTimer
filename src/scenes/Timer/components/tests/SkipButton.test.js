import React from 'react'
import * as redux from 'react-redux'
import { createShallow } from '@material-ui/core/test-utils'
import IconButton from '@material-ui/core/IconButton'
import { SkipButton } from '../SkipButton'
import * as timerActions from '../../data/timer/actions'

describe('<SkipButton />', () => {
  const shallow = createShallow()
  const createWrapper = () => {
    return shallow(<SkipButton />)
  }

  const dispatchMocked = jest.fn()
  jest.spyOn(redux, 'useDispatch').mockImplementation(() => dispatchMocked)

  const createStore = () => {
    const store = {
      timer: { timeLeft: { minutes: 25, seconds: 0 } },
      settings: {},
    }

    jest
      .spyOn(redux, 'useSelector')
      .mockImplementation((callback) => callback(store))

    return store
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('should render <SkipButton /> correctly', () => {
    createStore()
    expect(createWrapper()).toMatchSnapshot()
  })

  test('should handle skip button click', () => {
    const setNextTimerMocked = jest
      .spyOn(timerActions, 'setNextTimer')
      .mockImplementation(() => jest.fn())

    createStore()

    createWrapper().find(IconButton).simulate('click')
    expect(setNextTimerMocked).toHaveBeenCalled()
  })
})
