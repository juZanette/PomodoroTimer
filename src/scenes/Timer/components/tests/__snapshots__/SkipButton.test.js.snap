// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkipButton /> should render <SkipButton /> correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SkipButton>
    <WithStyles(ForwardRef(IconButton))
      aria-label="Skip current timer"
      disabled={false}
      onClick={[Function]}
    >
      <ForwardRef(IconButton)
        aria-label="Skip current timer"
        classes={
          Object {
            "colorInherit": "MuiIconButton-colorInherit",
            "colorPrimary": "MuiIconButton-colorPrimary",
            "colorSecondary": "MuiIconButton-colorSecondary",
            "disabled": "Mui-disabled",
            "edgeEnd": "MuiIconButton-edgeEnd",
            "edgeStart": "MuiIconButton-edgeStart",
            "label": "MuiIconButton-label",
            "root": "MuiIconButton-root",
            "sizeSmall": "MuiIconButton-sizeSmall",
          }
        }
        disabled={false}
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(ButtonBase))
          aria-label="Skip current timer"
          centerRipple={true}
          className="MuiIconButton-root"
          disabled={false}
          focusRipple={true}
          onClick={[Function]}
        >
          <ForwardRef(ButtonBase)
            aria-label="Skip current timer"
            centerRipple={true}
            className="MuiIconButton-root"
            classes={
              Object {
                "disabled": "Mui-disabled",
                "focusVisible": "Mui-focusVisible",
                "root": "MuiButtonBase-root",
              }
            }
            disabled={false}
            focusRipple={true}
            onClick={[Function]}
          >
            <button
              aria-label="Skip current timer"
              className="MuiButtonBase-root MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <ForwardRef>
                  <WithStyles(ForwardRef(SvgIcon))>
                    <ForwardRef(SvgIcon)
                      classes={
                        Object {
                          "colorAction": "MuiSvgIcon-colorAction",
                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                          "colorError": "MuiSvgIcon-colorError",
                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                          "root": "MuiSvgIcon-root",
                        }
                      }
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6 18l8.5-6L6 6v12zM16 6v12h2V6h-2z"
                        />
                      </svg>
                    </ForwardRef(SvgIcon)>
                  </WithStyles(ForwardRef(SvgIcon))>
                </ForwardRef>
              </span>
              <WithStyles(memo)
                center={true}
              >
                <ForwardRef(TouchRipple)
                  center={true}
                  classes={
                    Object {
                      "child": "MuiTouchRipple-child",
                      "childLeaving": "MuiTouchRipple-childLeaving",
                      "childPulsate": "MuiTouchRipple-childPulsate",
                      "ripple": "MuiTouchRipple-ripple",
                      "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                      "rippleVisible": "MuiTouchRipple-rippleVisible",
                      "root": "MuiTouchRipple-root",
                    }
                  }
                >
                  <span
                    className="MuiTouchRipple-root"
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      component={null}
                      exit={true}
                    />
                  </span>
                </ForwardRef(TouchRipple)>
              </WithStyles(memo)>
            </button>
          </ForwardRef(ButtonBase)>
        </WithStyles(ForwardRef(ButtonBase))>
      </ForwardRef(IconButton)>
    </WithStyles(ForwardRef(IconButton))>
  </SkipButton>
</Provider>
`;
